keyboards/darkproject/kd83a_ltd/kd83a_ltd.c

```diff
@@ -0,0 +1,152 @@
+    {0, CS7_SW11,  CS8_SW11,  CS9_SW11},  //  40, P
+    {0, CS7_SW12,  CS8_SW12,  CS9_SW12},  //  41, [
+    {1, CS1_SW8,   CS2_SW8,   CS3_SW8},   //  42, ]
+    {1, CS1_SW9,   CS2_SW9,   CS3_SW9},   //  43, "\\"
+    {1, CS4_SW3,   CS5_SW3,   CS6_SW3},   //  44, End
+  
+    {0, CS10_SW1,  CS11_SW1,  CS12_SW1},  //  45, Caps Lock
+    {0, CS10_SW2,  CS11_SW2,  CS12_SW2},  //  46, A
+    {0, CS10_SW3,  CS11_SW3,  CS12_SW3},  //  47, S
+    {0, CS10_SW4,  CS11_SW4,  CS12_SW4},  //  48, D
+    {0, CS10_SW5,  CS11_SW5,  CS12_SW5},  //  49, F
+    {0, CS10_SW6,  CS11_SW6,  CS12_SW6},  //  50, G
+    {0, CS10_SW7,  CS11_SW7,  CS12_SW7},  //  51, H
+    {0, CS10_SW8,  CS11_SW8,  CS12_SW8},  //  52, J
+    {0, CS10_SW9,  CS11_SW9,  CS12_SW9},  //  53, K
+    {0, CS10_SW10, CS11_SW10, CS12_SW10}, //  54, L
+    {0, CS10_SW11, CS11_SW11, CS12_SW11}, //  55, ;
+    {0, CS10_SW12, CS11_SW12, CS12_SW12}, //  56, '
+    {1, CS1_SW11,  CS2_SW11,  CS3_SW11},  //  57, Enter
+    {1, CS4_SW4,   CS5_SW4,   CS6_SW4},   //  58, PgUp
+   
+    {0, CS13_SW1,  CS14_SW1,  CS15_SW1},  //  59, Shift_L
+    {0, CS13_SW2,  CS14_SW2,  CS15_SW2},  //  60, Z
+    {0, CS13_SW3,  CS14_SW3,  CS15_SW3},  //  61, X
+    {0, CS13_SW4,  CS14_SW4,  CS15_SW4},  //  62, C
+    {0, CS13_SW5,  CS14_SW5,  CS15_SW5},  //  63, V
+    {0, CS13_SW6,  CS14_SW6,  CS15_SW6},  //  64, B
+    {0, CS13_SW7,  CS14_SW7,  CS15_SW7},  //  65, N
+    {0, CS13_SW8,  CS14_SW8,  CS15_SW8},  //  66, M
+    {0, CS13_SW9,  CS14_SW9,  CS15_SW9},  //  67, ,
+    {0, CS13_SW10, CS14_SW10, CS15_SW10}, //  68, .
+    {0, CS13_SW11, CS14_SW11, CS15_SW11}, //  69, /
+    {1, CS4_SW8,   CS5_SW8,   CS6_SW8},   //  70, Shift_R
+    {1, CS4_SW9,   CS5_SW9,   CS6_SW9},   //  71, Up
+    {1, CS4_SW7,   CS5_SW7,   CS6_SW7},   //  72, PgDn
+
+    {0, CS16_SW1,  CS17_SW1,  CS18_SW1},  //  73, Ctrl_L
+    {0, CS16_SW2,  CS17_SW2,  CS18_SW2},  //  74, Win_L
+    {0, CS16_SW3,  CS17_SW3,  CS18_SW3},  //  75, Alt_L
+    {0, CS16_SW6,  CS17_SW6,  CS18_SW6},  //  76, Space
+    {0, CS16_SW9,  CS17_SW9,  CS18_SW9},  //  77, Alt_R
+    {0, CS16_SW10, CS17_SW10, CS18_SW10}, //  78, FN
+    {0, CS16_SW12, CS17_SW12, CS18_SW12}, //  79, Ctrl_R
+    {1, CS4_SW10,  CS5_SW10,  CS6_SW10},  //  80, Left
+    {1, CS4_SW11,  CS5_SW11,  CS6_SW11},  //  81, Down
+    {1, CS10_SW5,  CS11_SW5,  CS12_SW5},  //  82, Right
+   
+    {1, CS10_SW10, CS11_SW10, CS12_SW10}, //  83, Encoder UP
+    {1, CS10_SW11, CS11_SW11, CS12_SW11}, //  84, Encoder DOWN
+};
+#endif
+
+
+#ifdef ENCODER_ENABLE
+
+bool encoder_update_kb(uint8_t index, bool clockwise) {
+
+    if (!encoder_update_user(index, clockwise)) { return false; }
+    if (index == 0) {
+        if (clockwise) {
+            tap_code(KC_VOLU);
+        } else {
+            tap_code(KC_VOLD);
+        }
+    }
+    return true;
+}
+#endif
+
+void led_init_ports(void) {
+  setPinOutput(C0);
+  setPinOutput(C14);
+  setPinOutput(C15);
+  writePinLow(C0);
+  writePinLow(C14);
```

Comment:
```suggestion
  setPinOutput(C15);
  writePinLow(C0);
```

Reply:
I must initialize caps lock pin (C14) here because this is override for the whole led pin init function and C14 pin won't be initialised anywhere else. I checked your variant and caps lock led does not work.